src/main/java/frc/robot/utils/PoseBuffer.java

```diff
@@ -0,0 +1,19 @@
+package frc.robot.utils;
+import java.util.ArrayDeque;
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.Timer;
+public class PoseBuffer {
```

Comment:
Also needs function to query an arbitrary timestamp, where we interpolate if it's within the range in the buffer, and return an error if the query is outside the range

Reply:
And this function should be tested.

Can we also check that these tests are run by GitHub ci?